src/code/z_lights.c

```diff
@@ -1,190 +1,392 @@
-        Lights_Remove(globalCtx, lightCtx, lightCtx->lightsHead);
-        lightCtx->lightsHead = lightCtx->lightsHead->next;
+void Lights_ClearBuf(GlobalContext* globalCtx, LightContext* lightCtx) {
+    while (lightCtx->head != NULL) {
+        Lights_Remove(globalCtx, lightCtx, lightCtx->head);
+        lightCtx->head = lightCtx->head->next;
     }
 }
 
-z_Light* Lights_Insert(GlobalContext* globalCtx, LightingContext* lightCtx, void* info) {
-    z_Light* light;
+LightNode* Lights_Insert(GlobalContext* globalCtx, LightContext* lightCtx, LightInfo* info) {
+    LightNode* node;
 
-    light = Lights_FindFreeSlot();
-    if (light != NULL) {
-        light->info = info;
-        light->prev = NULL;
-        light->next = lightCtx->lightsHead;
+    node = Lights_FindBufSlot();
 
-        if (lightCtx->lightsHead != NULL) {
-            lightCtx->lightsHead->prev = light;
+    if (node != NULL) {
+        node->info = info;
+        node->prev = NULL;
+        node->next = lightCtx->head;
+
+        if (lightCtx->head != NULL) {
+            lightCtx->head->prev = node;
         }
 
-        lightCtx->lightsHead = light;
+        lightCtx->head = node;
     }
 
-    return light;
+    return node;
 }
 
-void Lights_Remove(GlobalContext* globalCtx, LightingContext* lightCtx, z_Light* light) {
-    if (light != NULL) {
-        if (light->prev != NULL) {
-            light->prev->next = light->next;
+void Lights_Remove(GlobalContext* globalCtx, LightContext* lightCtx, LightNode* node) {
+    if (node != NULL) {
+        if (node->prev != NULL) {
+            node->prev->next = node->next;
         } else {
-            lightCtx->lightsHead = light->next;
+            lightCtx->head = node->next;
         }
 
-        if (light->next != NULL) {
-            light->next->prev = light->prev;
+        if (node->next != NULL) {
+            node->next->prev = node->prev;
         }
 
-        Lights_Free(light);
+        Lights_FreeNode(node);
+    }
+}
+
+// unused
+Lights* Lights_AllocAndDraw(GraphicsContext* gfxCtx, u8 ambientR, u8 ambientG, u8 ambientB, u8 numLights, u8 r, u8 g,
+                            u8 b, s8 x, s8 y, s8 z) {
+    Lights* lights;
+    s32 i;
+
+    lights = Graph_Alloc(gfxCtx, sizeof(Lights));
+
+    lights->l.a.l.col[0] = lights->l.a.l.colc[0] = ambientR;
+    lights->l.a.l.col[1] = lights->l.a.l.colc[1] = ambientG;
+    lights->l.a.l.col[2] = lights->l.a.l.colc[2] = ambientB;
+    lights->numLights = numLights;
+
+    for (i = 0; i < numLights; i++) {
+        lights->l.l[i].l.col[0] = lights->l.l[i].l.colc[0] = r;
+        lights->l.l[i].l.col[1] = lights->l.l[i].l.colc[1] = g;
+        lights->l.l[i].l.col[2] = lights->l.l[i].l.colc[2] = b;
+        lights->l.l[i].l.dir[0] = x;
+        lights->l.l[i].l.dir[1] = y;
+        lights->l.l[i].l.dir[2] = z;
+    }
+
+    Lights_Draw(lights, gfxCtx);
+
+    return lights;
+}
+
+Lights* Lights_AllocAndInit(GraphicsContext* gfxCtx, u8 r, u8 g, u8 b) {
```

Comment:
I defintiley see your reasoning why that makes sense here, but the wrapper is what gets used externally and is why I feel it should have the simple and effective name of New.
NewAmbient to me sounds like its only making a new ambient light, when its a new ambient + up to 7 other lights. I may be on board with a different name, but I really like Lights_New for this

Reply:
maybe we can go for something like Setup for AllocAndInit, as it allocates the Lightsn and sets the ambient color?
I dont think the name of Lights_New has to indicate that its setting the ambient color, it feels like more of an implementation detail to me

Reply:
How about _NewWithGlobalAmbient instead of _NewAmbient then? Since there aren't many arguments it can afford to have an explicit name imo. And I think it's important to not hide this part, since we wouldn't expect a completely "new" lights to already have the ambient set.

Reply:
so I decided to actually go ahead and make the system name changes for LightContext related functions.
It got me thinking, LightContext_NewLights could be the name for this function, and the fact that it sets the ambient color as that provided by light context can be an implementation detail.
While Lights_New is not a part of the LightContext system and can just be named that

What you think?

Reply:
I think it's fine. The ambient color might be more than an implementation detail, but it's built into the system anyway so it's probably supposed to be the "normal" way to create lights.